VocaDbModel/Service/VideoServices/VideoServiceBilibili.cs

```diff
@@ -61,37 +42,37 @@ public VideoServiceBilibili()
 				return VideoUrlParseResult.CreateOk(url, PVService.Bilibili, id, VideoTitleParseResult.Empty);

```

Comment:
I think we should use aid instead of id here. Can we do this after getting a BilibiliResponse?

Reply:
Not sure what you mean.

Reply:
Ok yeah, I see it now. Because here you're returning id, but below you're returning aid. Actually, I'm thinking, could you just always return the given id, either aid or bvid?

Reply:
> Are you talking about always returning aid instead of id (which may be bvid)?

Yes.

> And would you use the Encode/Decode code to get the ID?

No, we don't need to use the Encode/Decode code to get the ID ([I've already removed it](https://github.com/VocaDB/vocadb/pull/581/commits/46b725052ae075eec6c01f8c1c3d13e7098f1a5f)). Instead, we can use the `aid` and `bvid` fields included in a `BilibiliResponse`. That's why I want to move this if block after the try-catch block.

> Am I correct that this ID always includes the prefix (av or bv), so we can always figure out whether it's aid or bvid?

The `id` variable may be either aid or bvid as you said. If the `id` variable is an aid, then it is just an integer, that is, without the `av` prefix. For example, `6009789`. If the `id` variable is a bvid, then it starts with the `BV` prefix. For example, `BV1Qs411k7Qv`.

Reply:
Ideally, we would avoid any web requests if getTitle is false. It's a performance optimization, because sometimes we only need the ID and not metadata, and the request might be made in a loop for multiple PVs. But since Bilibili isn't used as much as YouTube or Nico Nico, I guess it's not a big issue. Some other parsers such as Piapro also make a web request for parsing ID, because they have to. If you're going that route, then you might as well just remove this if check, right?